src/Desktop.Analyzers/Core/TypesShouldNotExtendCertainBaseTypes.cs

```diff
@@ -65,10 +65,10 @@ private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
 
                         if (badBaseTypes.Contains(namedTypeSymbol.BaseType))
                         {
-                            var baseTypeName = namedTypeSymbol.BaseType.ToDisplayString();
+                            string baseTypeName = namedTypeSymbol.BaseType.ToDisplayString();
                             Debug.Assert(s_badBaseTypesToMessage.ContainsKey(baseTypeName));
-                            var message = string.Format(s_badBaseTypesToMessage[baseTypeName], namedTypeSymbol.ToDisplayString(), baseTypeName);
-                            var diagnostic = Diagnostic.Create(Rule, namedTypeSymbol.Locations.First(), namedTypeSymbol.Locations.Skip(1), message);
+                            string message = string.Format(s_badBaseTypesToMessage[baseTypeName], namedTypeSymbol.ToDisplayString(), baseTypeName);
+                            Diagnostic diagnostic = Diagnostic.Create(Rule, namedTypeSymbol.Locations.First(), namedTypeSymbol.Locations.Skip(1), message);
```

Comment:
It would be nice if the rule assumed that `Foo.Create` creates a `Foo`, but I guess that relies on convention as opposed to language rules.